### ISSUE TITLE
Scrollback patch request

### ISSUE THREAD

**Issue by jackpeters667** [ashish-yadav11/st#2]
Sweet work buddy. Would you be able to provide a scrollback patch here or on the suckless site? So everyone can share your awesomeness?

**Comment by ashish-yadav11** [comment]
I will but it can take a while.

**Comment by sewnie** [comment]
waits very patiently for it ( amazing work btw! )

**Comment by ashish-yadav11** [comment]
I haven't had time yet to extract the patch. But thanks to [koetemagie](https://github.com/BeyondMagic), you can find one [here](https://github.com/BeyondMagic/st/blob/master/patches/columns-rows-reflow-st-unpatched.patch). I haven't looked at the patch yet. If required I will refine it and update here.

**Comment by BeyondMagic** [comment]
> If required I will refine it and update here.

It would be better, since you made it, you know what should and shouldn't be on the patch.

**Comment by sewnie** [comment]
https://gist.github.com/wael444/bc44d877d0f6fe96411c49d265dbe82d
is this good?

**Comment by ashish-yadav11** [comment]
There are some unnecessary parts but otherwise I don't see any problem.

**Comment by ashish-yadav11** [comment]
The STRESCARG thing isn't required. I have separately provided it in "title parsing fix" titled patch on suckless' site.

**Comment by sewnie** [comment]
> The STRESCARG thing isn't required. I have separately provided it in "title parsing fix" titled patch on suckless' site.

oh. well it was included in the fused patch that was by BeyondMagic
well could you modify the diff and remove it? or is it easy to just replace all lines that look like STRESCARGEST to strescseq.args?

**Comment by ashish-yadav11** [comment]
yeah just remove STRESCARG and every change in strhandle function.

**Comment by sewnie** [comment]
`STRESCARGREST` to `strescseq.args` and
`STRESCARGJUST` to what?
it is only used in 
`		par = (narg = strescseq.narg) ? atoi(STRESCARGJUST(0)) : 0;
`

**Comment by ashish-yadav11** [comment]
remove every change in strhandle function. Keep it vanilla and remove the definition of STRESC... at the top.

**Comment by sewnie** [comment]
could you check on the new revision of the gist?
i also assume that the patch name should be added to the `scrollback` patch section with the name
`st-scrollback-column-reflow-20220410-ef05519.diff`

**Comment by BeyondMagic** [comment]
The `scrollback` is maintained by another person, and I'm pretty sure how the scroll works in this patch is too different to add there (In my opinion, obviously).

**Comment by sewnie** [comment]
honestly, i just formatted the patch
the main authors or whoever is related can choose what the patch name and description should be
if i were to submit the package id have to wait on @ashish-yadav11 's call on if the patch is good to be submitted.
then make its name seen in my previous comment.

**Comment by sewnie** [comment]
for some reason, now my patch sets a broken title. 

my only fix was to patch in `title parsing fix`.

**Comment by ashish-yadav11** [comment]
remove the changes in strparse function.

**Comment by sewnie** [comment]
> remove the changes in strparse function.

well that fixed it. i didn't know there were still strparse changes, thanks!

also, is this still good to be submitted yet? or is there still some problems?

**Comment by ashish-yadav11** [comment]
You can submit if you want to. Do add a link to this repo in case of bugs or for support.

**Comment by sewnie** [comment]
> You can submit if you want to. Do add a link to this repo in case of bugs or for support.

alright.. i will just do a pretty basic description for that matter.
i will also update your patch `title parsing fix` since it doesnt apply cleanly

**Comment by sewnie** [comment]
a new scroll back patch was released. [st-scrollback-0.8.5](https://st.suckless.org/patches/scrollback/st-scrollback-0.8.5.diff), i did not base my patches over this new one.

https://gist.github.com/wael444/fcfe9fd27c7f0c75f45aa9275d0abca0
contains both the patch that is placed ontop of scrollback >=0.8.5 and another patch which fuses both scrollback 0.8.5 and column reflow. 

are they good?

btw. the patch will be named `column_reflow` just keep that in mind as i have no other name for it.

**Comment by ashish-yadav11** [comment]
If after patching you get the same code, then it must be fine.

**Comment by sewnie** [comment]
sorry for the constant comments, i will be putting this under the `scrollback` patch index.md as this the original patch requires to patch it on top of `scrollback` 0.8.5 as it makes sense for it to be added there, unless [BeyondMagic](https://github.com/BeyondMagic)'s comment is correct and this should be placed under a different patch index as of how different it is.

i also believe by adding this to the `scrollback` patch page it'll be much easier for users to find this patch.

**Comment by sewnie** [comment]
it breaks [st-scrollback-mouse-20220127-2c5edf2.diff](https://st.suckless.org/patches/scrollback/st-scrollback-mouse-20220127-2c5edf2.diff), it no longer scrolls.

edit: turns out the patch is actually malformed and is missing some things...

**Comment by sewnie** [comment]
https://st.suckless.org/patches/scrollback/

well, this should be closed now i guess
